src/core/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,136 @@
+  <Color key={8} color='#008BE1' name='BLUE COLA' />,
+  <Color key={9} color='#30B0FF' name='FANTASY CONSOLE SKY' />,
+  <Color key={10} color='#58BFFF' name='HELLO SUMMER' />,
+  <Color key={11} color='#7FCEFF' name='CHROMIS DAMSEL BLUE' />,
+  <Color key={12} color='#A6DDFF' name='DROPLET' />,
+  <Color key={13} color='#CDECFF' name='LUCID DREAMS' />,
+  <Color key={14} color='#E5F5FD' name='KODAMA WHITE' />,
+  <Color key={15} color='#010200' name='REGISTRATION BLACK' />,
+  <Color key={16} color='#122306' name='YUZU SOY' />,
+  <Color key={17} color='#23450b' name='FOREST GREEN' />,
+  <Color key={18} color='#346711' name='TATZELWURM GREEN' />,
+  <Color key={19} color='#458816' name='CHLOROPHYLL' />,
+  <Color key={20} color='#56AA1C' name='PLASTIC PINES' />,
+  <Color key={21} color='#5FBB1F' name='FIELD GREEN' />,
+  <Color key={22} color='#67CC22' name='GREEN HIGH' />,
+  <Color key={23} color='#91E458' name='LILLIPUTIAN LIME' />,
+  <Color key={24} color='#B2EC8B' name='GREEN DAY' />,
+  <Color key={25} color='#D4F4BD' name='TEA GREEN' />,
+  <Color key={26} color='#EEF6E8' name='VERDE PASTEL' />,
+  <Color key={27} color='#9BA5AF' name='SERYI GREY' />,
+  <Color key={28} color='#CF0000' name='RED EPIPHYLLUM' />,
+  <Color key={29} color='#FF6300' name='SAFETY ORANGE' />,
+  <Color key={30} color='#FFC335' name='RISE-N-SHINE' />,
+];
+
+export type ColorPickerProps = {
+  /**
+   * List of colors to show in color palette
+   * Can pass in strings of the color values for simple input or a list of Colors to customize onColorClick handler etc.
+   */
+  colors?: ReactNode[];
+  /**
+   * Style of the color picker - basic or advanced.
+   * @default 'basic'
+   */
+  type?: 'basic' | 'advanced';
+  /**
+   * Index of the selected color on the color palette.
+   */
+  activeColorIndex?: number;
+  /**
+   * Handler for clicking a color
+   * This will be applied to all colors in the palette unless specifically passed in to a specific color in the colors list
+   */
+  onColorClicked?: (index: number) => void;
+  /**
+   * Any custom nodes that will be added within the color palette
+   */
+  children?: React.ReactNode;
+};
+
+export const ColorPicker = (props: ColorPickerProps) => {
+  const {
+    colors,
+    type = 'basic',
+    activeColorIndex,
+    onColorClicked,
+    children,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  // Set active color swatch
+  const [currentActiveIndex, setCurrentActiveIndex] = React.useState(
+    activeColorIndex,
+  );
+  const onColorChanged = (index: number) => {
+    setCurrentActiveIndex(index);
+  };
+
+  return (
+    <div
+      className={cx('iui-color-picker', {
+        [`iui-${type}`]: type !== 'basic',
+      })}
+      {...rest}
+    >
+      <div className='iui-color-palette'>
+        {colors &&
+          colors.length > 0 &&
+          colors.map((color, index) => {
+            const onClick = () => {
+              if (onColorClicked != null) {
+                onColorClicked(index);
```

Comment:
you could have optional chaining in one liner like `onColorClicked?.(index)`

Reply:
Changed